cardano-node/src/Cardano/Tracing/Tracers.hs

```diff
@@ -665,6 +670,63 @@ mpTracer :: ( Show (ApplyTxErr blk)
          => TraceSelection -> Trace IO Text -> Tracer IO (TraceEventMempool blk)
 mpTracer tc tr = annotateSeverity $ toLogObject' (traceVerbosity tc) tr
 
+--------------------------------------------------------------------------------
+-- ForgeState Tracers
+--------------------------------------------------------------------------------
+
+forgeStateMetricsTraceTransformer
+  :: forall a blk. HasKESMetricsData blk
+  => ProtocolInfo IO blk
+  -> Trace IO a
+  -> Tracer IO (ForgeState blk)
+forgeStateMetricsTraceTransformer protoInfo tr = Tracer $ \forgeState -> do
+    case getKESMetricsData protoInfo forgeState of
+      NoKESMetricsData -> pure ()
+      TPraosKESMetricsData _ _ NoOperationalCertConfigured -> pure ()
+      TPraosKESMetricsData kesPeriodOfKey
+                           (MaxKESEvolutions maxKesEvos)
+                           (OperationalCertStartKESPeriod oCertStartKesPeriod) -> do
+        let tr' = appendName "metrics" tr
+
+            -- The KES period of the hot key is relative to the start KES
+            -- period of the operational certificate.
+            currentKesPeriod = oCertStartKesPeriod + kesPeriodOfKey
+
+            oCertExpiryKesPeriod = oCertStartKesPeriod + fromIntegral maxKesEvos
+
+            kesPeriodsUntilExpiry =
+              max 0 (oCertExpiryKesPeriod - currentKesPeriod)
+
+            logValue :: LOContent a
+            logValue =
+              LogValue "remainingKESPeriods"
+                $ PureI
+                $ fromIntegral kesPeriodsUntilExpiry
+
+        meta <- mkLOMeta Critical Confidential
+        traceNamedObject tr' (meta, logValue)
```

Comment:
Do we not also want to trace the other values? The `oCertStartKesPeriod`, `oCertExpiryKesPeriod` and `currentKesPeriod`? Might as well.

Reply:
yes please :)